### PR TITLE
APInf Test Environment Setup Guide along with Test Scripts

### PR DESCRIPTION
**Pull request by kumargs**

### CODE REVIEW

**Comment by bajiat** [comment]
@apinf/developers Could one of you check the instructions and see if you can run the tests (in the zip) locally from your laptop? The tests are currently running against nightly. (I have tested instructions for Windows and was able to run the tests in my environment)

**Comment by jykae** [comment]
@kumargs please improve the instructions. We can make it shorter.
I can't see why it is necessary to install Java when we run JavaScript tests?
"npm install" is able to install all required stuff without explicitly installing each when you provide "package.json" file with dependencies.

On Mac OSX,
I created "test" folder, where I put unzipped files. Ran "npm install" then "npm install -g mocha"

Got "Error: The ChromeDriver could not be found on the current PATH. Please download the latest version of the ChromeDriver from http://chromedriver.storage.googleapis.com/index.html and ensure it can be found on your PATH."

Solved by downloading driver http://chromedriver.storage.googleapis.com/index.html and placing it in folder in PATH.

Then ran again "mocha" outside "test" folder, login and registration ran OK.

**Comment by jykae** [comment]
You could add all files in "test" folder already here in PR. Also unzip the files, easier for PR review in Github.

**Comment by brylie** [comment]
Yes, @kumargs please provide the test files in a format that we can track via Git. E.g. plain text files rather than a zip archive.

**Comment by kumargs** [comment]
Sprint 25 Scripts added. Along Installation Instructions were updated, created test folder and placed the files directly and removed the zip file.

**Comment by brylie** [comment]
@jykae please review.

**Comment by jykae** [comment]
@kumargs APInf_TestEnv_Setup_Guide.md is messed, includes some Git conflict lines, please clean up.

**Comment by jykae** [comment]
@apinf/developers could someone else take a look at this, try the tests and give @kumargs feedback, and merge in when clear problems are fixed? As I am going for summer holiday.

File: APInf_TestEnv_Setup_Guide.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Goto Terminal and execute the below command.
+```sh
+$ sudo apt-get install npm
+```
+If you already have it, then no action needed. 
+
+### Installation of Chrome Driver
+
+- Create a folder called *bin* under HOME Directory
+- Download [Chrome Driver]
```
Comment (brylie) [inline APInf_TestEnv_Setup_Guide.md:46]: Add link to Chrome Driver download.

**Comment by kumargs** [comment]
image

@brylie i am able to see the link to download the Chrome Driver. Please find attached image.  Have added those reference links to bottom of the page. 

image

http://chromedriver.storage.googleapis.com/index.html?path=2.22/ 

Please do let me know if you need more information

**Comment by brylie** [comment]
Ah, OK. Thanks for the clarification. I hadn't seen that syntax for links.

**Comment by brylie** [comment]
We can merge this code as it is. However, I am a bit concerned about the readability of the code. It is quite dense with no comments. Variable names are generally meaningful though, so that is good. Also, there are at least a couple of lines that could be broken into multiple steps for clarity and 'safety'. I can make notes describing the 'coding standards' we use when collaborating on code.

**Comment by brylie** [comment]
We have general notes about our [code guidelines](https://github.com/apinf/api-umbrella-dashboard/blob/develop/CONTRIBUTING.md#contributing-code) in the Apinf CONTRIBUTING file. In general, those guidelines should apply to our test code.

**Comment by kumargs** [comment]
@brylie Sure, thanks for sharing the code guidelines. Will include those comments as part of next sprint, as due to lots of layout changes happened, fix the scripts with the new layouts, making most commonly used functions to common file.

**Comment by brylie** [comment]
OK, this has been merged.

**Pull request merged**